### PR TITLE
Add connection args for pgvector vector store

### PR DESCRIPTION
**Pull request by aubinmazet**
- **Description:** sqlalchemy create_engine() does not take into account connect_args which are mandatory for managed PGSQL instances on cloud providers (ssl_context for example). 
Also re-enabled create_vector_extension at post_init for using pgvector class seamlessly
- **Tag maintainer:** @baskaryan, @eyurtsev, @hwchase17.

### CODE REVIEW

**Comment by Sbargaoui** [comment]
** Re-enabled create_vector_extension at post_init while raising an Exception if the Postgres instance can't create the extension to stop going through what's next

**Comment by eyurtsev** [comment]
note to self -- compare w/ https://github.com/langchain-ai/langchain/pull/12051

File: libs/langchain/langchain/vectorstores/pgvector.py [hunk 1]
```diff
@@ -108,6 +109,7 @@ def __init__(
         self,
         connection_string: str,
         embedding_function: Embeddings,
+        engine_args: Optional[dict[str, Any]] = None,
```
Comment (eyurtsev) [inline libs/langchain/langchain/vectorstores/pgvector.py:112]: @aubinmazet this is a breaking change since users may have been invoking this positionally, could we add at end of list and use a `*` preceding it?

File: libs/langchain/tests/integration_tests/vectorstores/test_pgvector.py [hunk 2]
```diff
@@ -44,6 +44,7 @@ def test_pgvector() -> None:
         embedding=FakeEmbeddingsWithAdaDimension(),
         connection_string=CONNECTION_STRING,
         pre_delete_collection=True,
+        engine_args={"pool_recycle": 3600},
```
Comment (eyurtsev) [inline libs/langchain/tests/integration_tests/vectorstores/test_pgvector.py:47]: Why is this needed in the tests?

Comment (Sbargaoui) [reply libs/langchain/tests/integration_tests/vectorstores/test_pgvector.py]: Added for testing the use of the new argument, the latest commit reverts the change.

File: libs/langchain/langchain/vectorstores/pgvector.py (hunk 1 above)
Comment (Sbargaoui) [reply libs/langchain/langchain/vectorstores/pgvector.py]: Updated the argument to a  **kwargs syntax
